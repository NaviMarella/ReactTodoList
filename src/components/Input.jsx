import React, { Component } from "react";
import PropTypes from 'prop-types';
import "./Input.css";

class Input extends Component{
    handleInputChange = (e) => {
        const { handleInput } = this.props;
        handleInput(e.target.value);
    }

    render(){
        const { type, value, maxLength, placeholder } = this.props;
        return (
            <div className="input_container">
                <input type={type} maxLength={maxLength} value={value} placeholder={placeholder} onChange={this.handleInputChange} />
            </div>
        )
    }
}

Input.defaultProps = {
    type: "text",
    maxLength: 50,
    placeholder: "Take a Note"
}

Input.propTypes = {
    placeholder: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    maxLength: PropTypes.number
};


export default Input;