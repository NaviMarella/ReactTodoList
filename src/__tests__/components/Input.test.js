import { shallow } from 'enzyme';
import React from 'react';
import Input from './../../components/Input';

describe('Input Component', () => {
  test('render snapshot', () => {
      let props = {
        placeholder: "Add a Note",
        type: "text",
        value: "Input Text",
        handleInput: jest.fn()
      }

    const event = {
        preventDefault: jest.fn(),
        target: { value: 'text' }
    };
    const wrapper = shallow(<Input {...props} />);
    wrapper.find('input').simulate('change', event);
    expect(props.handleInput).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
