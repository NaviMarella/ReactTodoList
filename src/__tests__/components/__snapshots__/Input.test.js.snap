// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    handleInput={
      [MockFunction] {
        "calls": Array [
          Array [
            "text",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    maxLength={50}
    placeholder="Add a Note"
    type="text"
    value="Input Text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        maxLength={50}
        onChange={[Function]}
        placeholder="Add a Note"
        type="text"
        value="Input Text"
      />,
      "className": "input_container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "maxLength": 50,
        "onChange": [Function],
        "placeholder": "Add a Note",
        "type": "text",
        "value": "Input Text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          maxLength={50}
          onChange={[Function]}
          placeholder="Add a Note"
          type="text"
          value="Input Text"
        />,
        "className": "input_container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "maxLength": 50,
          "onChange": [Function],
          "placeholder": "Add a Note",
          "type": "text",
          "value": "Input Text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
